dask/dataframe/io/parquet/arrow.py

```diff
@@ -1783,7 +1785,7 @@ def _gather_metadata(
                     schema = metadata.schema.to_arrow_schema()
             else:
                 for piece, fn in zip(dataset.pieces, fns):
-                    md = piece.get_metadata()
+                    md = cls.collect_file_metadata(piece.path, fs, None)
```

Comment:
@rjzamora it sounds like you have a better solution for this upstream?

Reply:
Maybe you are referring to #7295 ? (Which fixes a different problem with the `create_metadata_file` code path)

Reply:
ah ok. So does this seem like a reasonable solution? The issue was that pyarrow was doing `fs.open` in `get_metadata` and then never closing.

Reply:
Ah, that makes sense.  In that case, this looks like a good solution - Thanks for addressing this!  
 
The only other change that might make sense is to take advantage of the fact that `collect_file_metadata` can handle the `set_file_path` update if we pass `fn` as the last argument.  The one complication is that we would need to pass `piece.path.replace(base + fs.sep, "")` in the case that `fn_partitioned` is True.  I don't fell strongly one way or the other, but if you use `collect_file_metadata` for the file-path update, you can remove the `set_file_path` logic below.